lib/bolt/config/validator.rb

```diff
@@ -22,23 +25,36 @@ def validate(data, schema, location = nil)
           next unless schema.key?(key)
 
           @path.push(key)
+
+          warn_deprecated(schema[key], location)
           validate_value(value, schema[key])
         ensure
           @path.pop
         end
 
-        raise_error(location)
+        raise_error
+      end
+
+      # Adds a warning if the given option is deprecated.
+      #
+      def warn_deprecated(definition, location)
+        if definition.key?(:_deprecation)
+          message  = "Option '#{path}' "
+          message += "at #{location} " if location
+          message += "is deprecated. #{definition[:_deprecation]}"
+          @warnings << message
+        end
       end
 
       # Raises a ValidationError if there are any errors. All error messages
       # created during validation are concatenated into a single error
       # message.
       #
-      private def raise_error(location)
+      private def raise_error
         return unless @errors.any?
 
         message = "Invalid configuration"
-        message += " at #{location}" if location
+        message += " at #{@location}" if @location
```

Comment:
Where does this get set? This results in the location never being available:
```
lucy ~/githubs/bolt ((eec7edc...)) $ be bolt command run hostname -t docker
Invalid configuration:
  Value at 'color' must be of type Boolean
lucy ~/githubs/bolt ((eec7edc...)) $ git checkout main
Previous HEAD position was eec7edc... (GH-2363) Deprecate inventoryfile config option
Switched to branch 'main'
Your branch is up-to-date with 'puppet/main'.
lucy ~/githubs/bolt (main) $ be bolt command run hostname -t docker
Invalid configuration at /home/lucy/githubs/bolt/bolt-project.yaml:
  Value at 'color' must be of type Boolean
```

Reply:
Oops. That was leftover from looking at making `location` an instance var. Missed changing it back here. 😬